### DISCUSSION TITLE
Serving static files (windows/linux) from disk

### DISCUSSION THREAD

**Discussion by ThraaxSession** [micronaut-projects/micronaut-core#8173] [Q&A] [answered]
Hi together,

I would like to have the possibility to download my web-ui lazily. Therefore I want to serve it somewhere from the disk. It should work from Windows & *nix. 

I tried following on Windows (11). I was hoping the tilde was resolved. Any ideas how I get it work?

```
spaceup.home=~/.spaceup
micronaut.router.static-resources.default.mapping=/test/**
micronaut.router.static-resources.default.enabled=true
micronaut.router.static-resources.default.paths=file:${spaceup.home}/static
```

**Comment by ThraaxSession** [comment]
@graemerocher my favorite Micronaut developer. Do you have any idea?

**Reply by yawkat** [reply]
you could try the `${HOME}` env variable but it's not great for cross-platform

**Comment by alvarosanchez** [comment, marked as answer]
The `~` character is being expanded by shells, but outside a shell process it's meaningless.

This should work, though:

```properties
spaceup.home=${user.home}/.spaceup
```

**Reply by ThraaxSession** [reply]
Awesome! Thank you. :)
